main.go

```diff
@@ -55,7 +56,7 @@ func init() {
 	checkGolangRuntimeVersion()
 
 	// Check for the environment early on and gracefuly report.
-	_, err := userCurrent()
+	_, err := user.Current()
```

Comment:
Instead of checking if we are able to determine current user, we should check for configFolder is set or not. Config folder can be auto-determined via home dir (~/.minio) or custom. Check should only bail out if config folder is not set.  It should complain about requiring config folder. This check itself should be introduced as syntax check for main.